### PR TITLE
Empty string in smart contract

### LINKED ISSUE #5090: Encoding of uninitalized string property of a contract failed
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

For a contract which have a string property but not intialized, the returning value cause an encoding error. 

> AbiError: Parameter decoding error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.

### Expected Behavior

Such scenario should return into empty string. 

### Steps to Reproduce

The solidity contract. 

```
contract MyContract {
	string private myAttribute;

	function getAttr() public view returns (string memory) {
		return myAttribute;
	}
}
```

The web3 code. 

```
let contract = new Contract(MyContractAbi);
contract = await contract.deploy().send(....);

await contract.methods.getAttr().call();
```

The following scenario can also be reproduced with `Greeter.sol` contract we have. 

### Web3.js Version

4.x

### Environment

_No response_

### Anything Else?

_No response_

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #5090

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

**Review by avkos — COMMENTED** [review]
I didn't see any fixes. Is it just working?

**Comment by nikoulai** [comment]
> I didn't see any fixes. Is it just working?

yes, I guess it's an older issue and something changed in between.

File: packages/web3-eth-contract/test/integration/contract_empty_string.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { Contract } from '../../src';
+import { getSystemTestProvider, createTempAccount } from '../fixtures/system_test_utils';
+import { MyContractAbi, MyContractBytecode } from '../fixtures/MyContract';
+
+describe('contract', () => {
+	describe('clone', () => {
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-contract/test/integration/contract_empty_string.test.ts]: Maybe something more informative would be helpful.
The following is just a suggestion. And you may find something better :smile: 
```suggestion
describe('contract's function's calls', () => {
	describe('encoding edge cases', () => {
```

**Review by Muhammad-Altabba — APPROVED** [review]

**Review by avkos — APPROVED** [review]
Approved. But need to check why tests fail.

**Review by jdevcs — APPROVED** [review]
